### PR TITLE
py-pdal: add new port, version 3.2.3

### PR DESCRIPTION
**Pull request by nilason**
#### Description

Add new Python port `py-pdal`, which provides Python interface to the `PDAL` port.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.5 21G531 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-pdal/Portfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    control Python interaction with PDAL. Additionally, you\
+                    can use it to fetch schema and metadata from PDAL operations.
+
+homepage            https://www.pdal.io
+
+checksums           rmd160  3167d2d4181d7671bbc9afb9b3512d8e1d44b5f6 \
+                    sha256  0bc2cb113f4df45b239dd54aedb8026478349db2daa5c98569994308d0563807 \
+                    size    246729
+
+python.versions     37 38 39 310
```
[the remark this answers is not in the document]
Comment (nilason) [reply python/py-pdal/Portfile:27]: Strange. Here is a larger chunk of the log:

```
--->  Building py311-pdal
Executing:  cd "/opt/local/var/macports/build/_Users_nilason_ports_python_py-pdal/py311-pdal/work/pdal-3.2.3" && /opt/local/Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_Users_nilason_ports_python_py-pdal/py311-pdal/work 
* Getting build dependencies for wheel...
running egg_info
writing pdal.egg-info/PKG-INFO
writing dependency_links to pdal.egg-info/dependency_links.txt
writing requirements to pdal.egg-info/requires.txt
writing top-level names to pdal.egg-info/top_level.txt
listing git files failed - pretending there aren't any
reading manifest file 'pdal.egg-info/SOURCES.txt'
adding license file 'LICENSE.txt'
writing manifest file 'pdal.egg-info/SOURCES.txt'

ERROR Missing dependencies:
	pybind11[global]
	ninja
	cmake>=3.11
Command failed:  cd "/opt/local/var/macports/build/_Users_nilason_ports_python_py-pdal/py311-pdal/work/pdal-3.2.3" && /opt/local/Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_Users_nilason_ports_python_py-pdal/py311-pdal/work 
```

If I manually execute the failed command it succeeds.

[the remark this answers is not in the document]
Comment (nilason) [reply python/py-pdal/Portfile]: The problem with 3.11 was due to  python.pep517, now fixed.

File: python/py-pdal/Portfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-pdal
+python.rootname     pdal
```
Comment (reneeotten) [inline python/py-pdal/Portfile:7]: this is the default set by the python PG so not needed

File: python/py-pdal/Portfile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+version             3.2.3
+revision            0
+
+categories-append   gis
+license             BSD
+maintainers         {@nilason} openmaintainer
```
Comment (reneeotten) [inline python/py-pdal/Portfile:13]: please add you obfuscated e-mail address as well (or remove the curly brackets)

File: python/py-pdal/Portfile [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    port:py${python.version}-scikit-build \
+                    port:pybind11 \
+                    port:ninja
+
+    depends_lib-append \
+                    port:PDAL \
+                    port:py${python.version}-numpy
+
+    depends_test-append \
+                    port:py${python.version}-meshio \
```
Comment (reneeotten) [inline python/py-pdal/Portfile:47]: Remove line continuation

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-pdal/Portfile (hunk 2 above)
Comment (nilason) [reply python/py-pdal/Portfile]: removed

File: python/py-pdal/Portfile (hunk 3 above)
Comment (nilason) [reply python/py-pdal/Portfile]: added

File: python/py-pdal/Portfile (hunk 4 above)
Comment (nilason) [reply python/py-pdal/Portfile]: removed

**Review by reneeotten — APPROVED** [review]

**Pull request merged**